actions/run-nextstrain-build-on-aws-batch/action.yaml

```diff
@@ -0,0 +1,68 @@
+name: Run Nextstrain build on AWS Batch
+description: >-
+  This GitHub Actions action is intended to be called by workflows in our other
+  repos when they need to run a nextstrain build on AWS Batch.
+
+inputs:
+  runtime-envvar-names:
+    description: >-
+      A list of environment variable names for environment variables that
+      should be passed to the build. The envvar names should be separated by
+      a space, e.g. "GITHUB_RUN_ID SLACK_TOKEN SLACK_CHANNELS"
+    type: string
+    required: false
+
+  cpus:
+    description: >-
+      Number of CPUs/cores/threads/jobs to utilize at once.
+      Will be passed to the `--cpus` option for `nextstrain build` and the
+      `--cores` option for Snakemake.
+    type: number
+    default: 4
+    required: false
+
+  memory:
+    description: >-
+      Amount of memory to make available to the build. Passed to the
```

Comment:
We should specify the string format to use here, which presumably is "you must specify the units" as the help for `--memory` states "Units of b, kb, mb, gb, kib, mib, gib are supported."

Reply:
(`--aws-batch-memory` is different -- "Amount of memory in MiB to request for job")

Reply:
Alternatively, since we use c5 instances, should we be picking the maximum memory allocated for a machine with the specified CPUs? Or vice versa?

Reply:
Ah, I purposely left out any text here that would repeat the help output from the Nextstrain CLI to prevent them from getting out of sync. I could update the description with a link to the [nextstrain build docs](https://docs.nextstrain.org/projects/cli/en/stable/commands/build/).

Reply:
> Alternatively, since we use c5 instances, should we be picking the maximum memory allocated for a machine with the specified CPUs? Or vice versa?

Our compute environments set the [allocation strategy](https://docs.aws.amazon.com/batch/latest/userguide/allocation-strategies.html) to "BEST_FIT" so AWS will try to use an instance that fits the requested the resources at the lowest cost. I've tried to set a job to use 70GiB when the max memory is 72GiB, but AWS still used the next instance up (see [Slack thread](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1674254476788549)). 

So we just have to estimate the minimum resources needed for a job and AWS will figure out what instance to use.

Reply:
> Ah, I purposely left out any text here that would repeat the help output from the Nextstrain CLI to prevent them from getting out of sync. 

If I'm correctly imagining where this action will take us, it's plausible that the vast majority of AWS jobs we run will originate from this action. For instance, imagine we have _n_ caller workflows which each specify `memory: 32gib` (or similar), and then the nextstrain CLI changes its interface to wanting memory specified in kb as an INT; we'd now have to change _n_ caller workflows rather than just this one called workflow. Using major version tagging for this action, and coupling this action to a major version pin of nextstrain CLI, should avoid this fear of things getting out of sync. 

Put a different way, I think one should be able to go from running a snakemake workflow locally (without the nextstrain CLI) to using this action to automate the workflow without needing any specific knowledge of the nextstrain CLI itself. There's discussion above about "aim[ing] for the runtime to be ~transparent", and I think we should aim for the usage of nextstrain CLI itself to be ~transparent.

Reply:
(oh boy, I was confused by the term [transparent](https://en.wikipedia.org/wiki/Transparency_(human%E2%80%93computer_interaction)) for a bit there 😅 )

My original intention for this action/workflow is to reduce the work needed to _set up_ the GitHub Action workflow, but still be very clear that this is basically a wrapper around the Nextstrain CLI. I guess I'm not sure what is the benefit of adding another layer of abstraction on top of the Nextstrain CLI.

Reply:
> this is basically a wrapper around the Nextstrain CLI

Fair enough - and this does have value - much better than having dozens of bespoke GitHub actions out there callling `--exec` and using envdir workarounds. I'd like to see this GitHub action become much more than just a wrapper, but this can happen in time. Assuming this PR remains a wrapper I'd still encourage the documentation ('description') for these inputs to be more self-explanatory rather than just pointing back to the CLI.

Reply:
@jameshadfield [wrote](https://github.com/nextstrain/.github/pull/39#discussion_r1206230420):

> Put a different way, I think one should be able to go from running a snakemake workflow locally (without the nextstrain CLI) to using this action to automate the workflow without needing any specific knowledge of the nextstrain CLI itself. There's discussion above about "aim[ing] for the runtime to be ~transparent", and I think we should aim for the usage of nextstrain CLI itself to be ~transparent.

Hmm. I don't understand (nor agree with, at least currently) this desire at all. What's your basis for it?